### PR TITLE
Migrate all gke-master jobs from Jenkins to Prow

### PR DESCRIPTION
**Pull request by krzyzacy**
#3266 did not add too much pressure to prow, 50 nodes should be able to handle all our jobs.

### CODE REVIEW

File: prow/config.yaml [hunk 1]
```diff
@@ -9949,23 +11206,22 @@ periodics:
       secret:
         defaultMode: 256
         secretName: ssh-key-secret
-
-- name: ci-kubernetes-e2e-gke-gpu
-  interval: 2h
+- interval: 6h
```
Comment (rmmh) [inline prow/config.yaml]: what's with all these changes?

Comment (krzyzacy) [inline prow/config.yaml]: argh, we are now trying to keep prow/config.yaml sorted.

I can split it into two commits.

Comment (rmmh) [inline prow/config.yaml]: OK.

Might be worth it to have a "prow config diff" tool somewhere, like we have with the old jenkins configs. :-)

Comment (fejta) [inline prow/config.yaml]: That's this file

**Comment by krzyzacy** [comment]
okay, moment of truth for https://github.com/kubernetes/test-infra/pull/3283

**Pull request merged**